oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/pipelined/NodeDocumentCodec.java

```diff
@@ -98,8 +99,7 @@ public Class<NodeDocument> getEncoderClass() {
         return NodeDocument.class;
     }
 
-    private Object readValue(BsonReader reader, String fieldName) {
-        estimatedSizeOfCurrentObject += 16 + fieldName.length();
+    private Object readValue(BsonReader reader, String fieldName, int[] estimatedSizeOfCurrentObject) {
```

Comment:
Question: 
Why is `estimatedSizeOfCurrentObject` an `int[]`? I get that we pass it around so that we can modify it as a side effect in `readValue`. And I'm not sure this is the best approach from a design perspective.

Reply:
I wanted to avoid creating a new class. But I agree it is not elegant. I have changed to use the MutableInt in Apache Commons, which is already in the classpath.